### PR TITLE
[autoscaler] Remove Fingerprint checking

### LINKED ISSUE #9131: Ray should remove ssh fingerprints from known_hosts when tearing down a cluster
### Describe your feature request
When Ray kills a node or tears down a cluster, it should remove the ssh fingerprints for the corresponding IP addresses from ~/.ssh/known_hosts. GCP often reuses IP addresses, so if you delete and reprovision a node, it is common for GCP to give you the same IP. Because it's a new VM, the fingerprint is going to be different than the last node's fingerprint. This triggers `WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!`, which ruins the current action. This issue probably hasn't been noticed until now because AWS rarely gives you the same IP. Currently I've been fixing this by manually deleting entries from known_hosts, but it would be nice if this happened automatically (e.g. by executing `ssh-keygen -R $DEST`). This should also happen if Ray notices that the cluster has been deleted through the GCP/AWS console.

### PR DESCRIPTION
**Pull request by henktillman**
## Related issue number

Closes #9131 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested (please justify below)

Tested locally by executing `ray up` to create a new GCP cluster, then checking to ensure that no IPs were added to `~/.ssh/known_hosts`.

### CODE REVIEW

**Comment by thomasdesr** [comment]
I wonder if this is gonna cause problems on windows? 

cc @mehrdadn: halp! Is adding a reference to `/dev/null` in the autoscaler SSH command going to cause problems on Windows?

**Comment by mehrdadn** [comment]
Yes! Please use `os.devnull` instead. (Though even that might fail in some cases, but it usually works.)

**Comment by mehrdadn** [comment]
@thomasdesr Oh wait, I just realized: in the case of SSH, it seems to treat `/dev/null` fine. In general though you want to use the value of `os.devnull`.

**Review by richardliaw — APPROVED** [review]

File: python/ray/autoscaler/updater.py [hunk 1]
```diff
@@ -169,7 +169,14 @@ def __init__(self, log_prefix, node_id, provider, auth_config,
...
         OPTS = [
             ("ConnectTimeout", "{}s".format(connect_timeout)),
+            # Supresses initial fingerprint verification.
             ("StrictHostKeyChecking", "no"),
+            # SSH IP and fingerprint pairs no longer added to known_hosts.
+            # This is to remove a "REMOTE HOST IDENTIFICATION HAS CHANGED"
+            # warning if a new node has the same IP as a previously
+            # deleted node, because the fingerprints will not match in
+            # that case.
+            ("UserKnownHostsFile", "/dev/null"),
```
Comment (richardliaw) [inline python/ray/autoscaler/updater.py:179]: should we use os.devnull to be safe?

**Pull request merged**